lua/lean/infoview/components.lua

```diff
@@ -191,4 +203,72 @@ function components.diagnostics(bufnr, line)
   return div
 end
 
+---@param t TaggedTextMsgEmbed
+---@param sess Subsession
+local function tagged_text_msg_embed(t, sess)
+  local div = html.Div:new({}, "", "code-with-infos")
+
+  if t.text ~= nil then
+    div:insert_div({}, t.text, "text")
+  elseif t.append ~= nil then
+    for _, s in ipairs(t.append) do
+      div:insert_new_div(tagged_text_msg_embed(s, sess))
+    end
+  elseif t.tag ~= nil then
+    local embed = t.tag[1]
+    if embed.expr ~= nil then
+      return code_with_infos(embed.expr, sess)
+    elseif embed.goal ~= nil then
+      return interactive_goal(embed.goal, sess)
+    elseif embed.lazyTrace ~= nil then
+      local click = function(tick, render_fn)
+        if render_fn then
+          div.divs = {}
+          div:insert_div({}, "loading...", "loading-message")
+          render_fn()
+        end
+
+        local indent = embed.lazyTrace[1]
+        local expanded, err = sess:msgToInteractive(embed.lazyTrace[3], indent)
+        if not tick:check() then return true end
+
+        if err then
+          print("RPC ERROR:", vim.inspect(err.code), vim.inspect(err.message))
+          return false
+        end
+
+        div.divs = {}
+        div.highlightable = false
+        div.tags.event.click = nil
+        div:insert_new_div(tagged_text_msg_embed(expanded, sess))
+        return true
+      end
+      div.highlightable = true
+      div.tags.event = { click = click }
+      div:insert_div({}, '...')
```

Comment:
I think we should have a corresponding `clear` event that sets it back to this state so that undo works here.

Reply:
Also, how can I get this collapsed trace to show so I can try it out?